### PR TITLE
Fix crash when sending events to client.js during hot restart

### PR DESCRIPTION
**Pull request by annagrin**
If an event is sent to the injected client during hot restart, we might encounter a race where the response is added to a closed sink and injected client stops working. 

Catch and ignore those errors so the hot restart can continue.

Closes: https://github.com/dart-lang/webdev/issues/1530

### CODE REVIEW

**Review by grouma — APPROVED** [review]

File: dwds/web/client.dart [hunk 1]
```diff
@@ -68,8 +68,15 @@ Future<void> main() {
...
     debugEventController.stream.listen((events) {
-      client.sink.add(jsonEncode(serializers.serialize(BatchedDebugEvents(
-          (b) => b.events = ListBuilder<DebugEvent>(events)))));
+      try {
+        client.sink.add(jsonEncode(serializers.serialize(BatchedDebugEvents(
+            (b) => b.events = ListBuilder<DebugEvent>(events)))));
+      } on StateError catch (_) {
+        // An error is propagated on a full page reload as Chrome presumably
+        // forces the SSE connection to close in a bad state.
+        print('Cannot send BatchedDEbugEvents($events). '
```
Comment (elliette) [inline dwds/web/client.dart:77]: Small nit: DEbug -> Debug

File: dwds/web/client.dart [hunk 2]
```diff
@@ -82,9 +89,16 @@ Future<void> main() {
     });
 
     emitRegisterEvent = allowInterop((String eventData) {
-      client.sink.add(jsonEncode(serializers.serialize(RegisterEvent((b) => b
-        ..timestamp = (DateTime.now().millisecondsSinceEpoch)
-        ..eventData = eventData))));
+      try {
+        client.sink.add(jsonEncode(serializers.serialize(RegisterEvent((b) => b
```
Comment (elliette) [inline dwds/web/client.dart:93]: Optional suggestion: Helper method that will take an event and call `client.sink.add(jsonEncode(serializers.serialize()))` with it, maybe called `_addToEventSink`.

Comment (annagrin) [reply dwds/web/client.dart]: Thanks, done!

File: dwds/web/client.dart (hunk 1 above)
Comment (annagrin) [reply dwds/web/client.dart]: This is removed now in changes for the comment below.

**Pull request merged**